### PR TITLE
[AMORO-1293][Improvement][CI]: Run maven checkstyle first in ci

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?
 
 Optimize the execution order of GitHub CI. First, execute checkstyle to expose stupid issues first.

## Brief change log

- Remove the configuration of scalastyle-maven-plugin. 
- Execute the mvn validate command in the CI workflow.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -588,32 +586,6 @@
                         </execution>
                     </executions>
                 </plugin>
-                <plugin>
-                    <groupId>org.scalastyle</groupId>
-                    <artifactId>scalastyle-maven-plugin</artifactId>
-                    <version>${maven-scalastyle-plugin.version}</version>
```
Comment (wangtaohz) [inline pom.xml:594]: Do we not need this plugin now?

File: .github/workflows/core-hadoop2-ci.yml [hunk 2]
```diff
@@ -35,8 +35,8 @@ jobs:
           distribution: 'temurin'
           cache: maven
 
-      - name: Print maven depend tree
-        run: mvn dependency:tree -B -ntp -Dhadoop=v2
+      - name: checkstyle first
```
Comment (zhoujinsong) [inline .github/workflows/core-hadoop2-ci.yml:38]: How about rename the name to `Validate checkstyle first`?

Comment (zhoujinsong) [reply .github/workflows/core-hadoop2-ci.yml:38]: BTW，do you think we need add this change to `trino-ci.yml` too?

File: pom.xml (hunk 1 above)
Comment (baiyangtx) [reply pom.xml:594]: no, we use spotless instead.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Review by wangtaohz — APPROVED** [review]
LGTM.

**Comment by wangtaohz** [comment]
close #1293

**Pull request merged**